pandas/core/config_init.py

```diff
@@ -455,12 +455,27 @@ def is_terminal() -> bool:
     ``future.infer_string`` is set to True.
 """
 
+
+def is_valid_string_storage(value):
+    legal_values = ["python", "pyarrow"]
+    if value not in legal_values:
+        msg = "Value must be one of python|pyarrow"
+        if value == "pyarrow_numpy":
+            # TODO: we can remove extra message after 3.0
+            msg += (
+                ". 'pyarrow_numpy' was specified, but this option should be "
+                "enabled using pandas.options.future.infer_string instead"
+            )
```

Comment:
Added this custom message to the error in case someone actually did end up using `pd.options.mode.string_storage = "pyarrow_numpy"` to point them to the right option to enable.